includes/api/class-wc-admin-rest-customers-controller.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * REST API Customers Controller
+ *
+ * Handles requests to /customers/*
+ *
+ * @package WooCommerce Admin/API
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Customers controller.
+ *
+ * @package WooCommerce Admin/API
+ * @extends WC_REST_Customers_Controller
+ */
+class WC_Admin_REST_Customers_Controller extends WC_REST_Customers_Controller {
+
+	// TODO Add support for guests here. See https://wp.me/p7bje6-1dM.
+
+	/**
+	 * Searches emails by partial search instead of a strict match.
+	 * See "search parameters" under https://codex.wordpress.org/Class_Reference/WP_User_Query.
+	 *
+	 * @param array $prepared_args Prepared search filter args from the customer endpoint.
+	 * @param array $request Request/query arguments.
+	 * @return array
+	 */
+	public static function update_search_filters( $prepared_args, $request ) {
+		if ( ! empty( $prepared_args['search_columns'] ) && in_array( 'user_email', $prepared_args['search_columns'] ) && empty( $request['name'] ) ) {
+			$prepared_args['search'] = '*' . $prepared_args['search'] . '*';
+		}
+
+		return $prepared_args;
+	}
+
+	/**
+	 * Get the query params for collections.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params() {
+		$params = parent::get_collection_params();
+		// Allow partial email matches.
```

Comment:
Optional: expand on this comment to explain that the partial match is being allowed by way of removing the email validation from the parameter.

Reply:
Done!